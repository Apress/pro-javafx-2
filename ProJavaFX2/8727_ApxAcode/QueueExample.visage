/*
 * @test
 * @run
 */

var q:Integer[];

function enqueue(i:Integer):Void {
  insert i into q;
  println("Enqueued {i}.  Size of q is {sizeof q} now.");
}

function dequeue():Integer {
  if (sizeof q == 0) {
    println("Size of q is {sizeof q} now. Returning -1.");
    return -1;
  } else {
    var i = q[0];
    delete q[0];
    println("Dequeued {i}.  Size of q is {sizeof q} now.");
    return i;
  }
}

enqueue(1);
enqueue(2);
enqueue(3);
println(dequeue());
println(dequeue());
enqueue(4);
println(dequeue());
println(dequeue());
println(dequeue());
