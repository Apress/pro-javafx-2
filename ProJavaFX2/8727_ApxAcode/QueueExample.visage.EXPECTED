Enqueued 1.  Size of q is 1 now.
Enqueued 2.  Size of q is 2 now.
Enqueued 3.  Size of q is 3 now.
Dequeued 1.  Size of q is 2 now.
1
Dequeued 2.  Size of q is 1 now.
2
Enqueued 4.  Size of q is 2 now.
Dequeued 3.  Size of q is 1 now.
3
Dequeued 4.  Size of q is 0 now.
4
Size of q is 0 now. Returning -1.
-1
